File: zig/difference-of-squares/difference_of_squares.zig
pub fn squareOfSum(number: usize) usize {
    const sum = number * (number + 1) / 2;
    return sum * sum;
}

pub fn sumOfSquares(number: usize) usize {
    return (2 * number + 1) * (number + 1) * number / 6;
}

pub fn differenceOfSquares(number: usize) usize {
    return squareOfSum(number) - sumOfSquares(number);
}

File: c/difference-of-squares/difference_of_squares.c
#include "difference_of_squares.h"

unsigned sum_of_squares(unsigned number) {
  return (2 * number + 1) * (number + 1) * number / 6;
}

unsigned square_of_sum(unsigned number) {
  // Gauss's formula
  const unsigned sum = number * (number + 1) / 2;
  return sum * sum;
}

unsigned difference_of_squares(unsigned number) {
  return square_of_sum(number) - sum_of_squares(number);
}
